CMakeLists.txt

```
@@ -194,9 +194,28 @@ function(mbed_generate_map_file target)
 endfunction()
 
 #
-# Generate executables
+# To validate profile support based on `target` link libaries.
 #
-function(mbed_generate_executable target)
+function(mbed_check_if_application_profile_supported target)
+    get_target_property(app_link_libraries ${target} LINK_LIBRARIES)
+    string(REGEX MATCH "mbed-baremetal"
+          output_string ${app_link_libraries})
+    if(${output_string} STREQUAL "mbed-baremetal")
+        if(NOT "bare-metal" IN_LIST MBED_TARGET_SUPPORTED_APPLICATION_PROFILES)
+            message(FATAL_ERROR
+                "Use full profile as baremetal profile is not supported for this target")
```

```suggestion
                "Use full profile as baremetal profile is not supported for this Mbed target")
```
Should we be clearer here because "target" can also imply CMake target?